common/src/main/java/io/novafoundation/nova/common/validation/FieldValidator.kt

```diff
@@ -19,6 +20,23 @@ abstract class MapFieldValidator : FieldValidator {
     override fun observe(inputStream: Flow<String>) = inputStream.map(::validate)
 }
 
+class CompoundFieldValidator(
+    private val validators: List<FieldValidator>
+) : FieldValidator {
+
+    constructor(vararg validators: FieldValidator) : this(validators.toList())
+
+    override fun observe(inputStream: Flow<String>): Flow<FieldValidationResult> {
+        // The negative side of this approach is that we run all validations despite one of them may be failed
```

Comment:
I think its fine given that validation produce not a single value but a stream of values